app/src/test/java/org/mozilla/fenix/home/recenttabs/view/RecentTabViewHolderTest.kt

```diff
@@ -33,18 +38,23 @@ class RecentTabViewHolderTest {
     fun setup() {
         view = LayoutInflater.from(testContext).inflate(RecentTabViewHolder.LAYOUT_ID, null)
         interactor = mockk(relaxed = true)
+        icons = mockk(relaxed = true)
+
+        every { icons.loadIntoView(view.recent_tab_icon, any()) } returns mockk()
     }
 
     @Test
-    fun `GIVEN a new recent tab on bind THEN set the title text`() {
-        RecentTabViewHolder(view, interactor).bindTab(tab)
+    fun `GIVEN a new recent tab on bind THEN set the title text and load the tab icon`() {
+        RecentTabViewHolder(view, interactor, icons).bindTab(tab)
 
         assertEquals(tab.content.title, view.recent_tab_title.text)
+
+        verify { icons.loadIntoView(view.recent_tab_icon, IconRequest(tab.content.url)) }
```

Comment:
nit: could use a year for when the tab.content.icon is not null i.e. the other branch of the if-else.